### PR TITLE
Move and rename files (2/4)

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Everything

### Description
This PR moves and renames some components:
- Editor was renamed to EditorPage
- PadOnlyView was renamed to DocumentReadOnlyPage
- DocumentIframe was renamed RenderIframe
- DocumentRenderPane was renamed to MarkdownDocument and moved to `components/render-page`
- Everything related to synced scrolling (hooks, types, etc.) was moved into `editor-page/synced-scroll`
- The hook `src/components/editor/document-renderer-pane/hooks/use-adapted-line-markers-callback.ts` has been deleted, because it isn't necessary anymore.

### Steps

- [x] Added implementation
- [ ] Added / updated tests
- [ ] Added / updated documentation
- [ ] Extended changelog
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: src/components/editor-page/editor-pane/one-dark.scss [hunk 1]
```diff
@@ -1,11 +1,7 @@
-/*
- * Copyright (c) 2015 Hikio - twitter.com/hik_io
- * SPDX-FileCopyrightText: 2015 Hikio - twitter.com/hik_io
- *
- * SPDX-License-Identifier: MIT
+/*!
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
  *
- * Atom One Dark Theme
- * GitHub https://github.com/hikio/brackets-one-dark
```
Comment (DerMolly) [inline src/components/editor-page/editor-pane/one-dark.scss:8]: This seems wrong.

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: src/components/editor-page/editor-pane/one-dark.scss (hunk 1 above)
Comment (mrdrogdrog) [reply src/components/editor-page/editor-pane/one-dark.scss]: That's right. I guess the jetbrains feature "update copyright" has overwritten it.

**Review by DerMolly — APPROVED** [review]

File: src/components/editor-page/renderer-pane/render-iframe.tsx [hunk 2]
```diff
@@ -10,13 +10,13 @@ import { useIsDarkModeActivated } from '../../../hooks/common/use-is-dark-mode-a
 import { ApplicationState } from '../../../redux'
 import { isTestMode } from '../../../utils/is-test-mode'
 import { IframeEditorToRendererCommunicator } from '../../render-page/iframe-editor-to-renderer-communicator'
+import { MarkdownDocumentProps } from '../../render-page/markdown-document'
 import { ImageDetails } from '../../render-page/rendering-message'
 import { ScrollState } from '../scroll/scroll-props'
```
Comment (ErikMichelson) [inline src/components/editor-page/renderer-pane/render-iframe.tsx:15]:
```suggestion
import { ScrollState } from '../synced-scroll/scroll-props'
```

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: src/components/editor-page/renderer-pane/render-iframe.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/editor-page/renderer-pane/render-iframe.tsx]: Rebased

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**